### DISCUSSION TITLE
traceln(attempt.getFeedback()) gives java.lang.IndexOutOfBoundsException

### DISCUSSION THREAD

**Discussion by jacobtw2** [the-anylogic-company/AnyLogic-Pypeline#65] [Q&A]
I have been using pycommunicator to run an optimization algorithm and then parse json results into the database

This has worked well and I can see the task completing successfully (i.e results appearing in the database)

When I call traceln(attempt.getFeedback()) the java.lang.IndexOutOfBoundsException error appears in the console. It does not appear red or cause the simulation to stop. It does show the series of method calls that lead to the error and it points to the traceln(attempt.getFeedback()) line

If i just call the method without trying to trace it then there is no error in the console

Any insight into what causes this?

**Comment by t-wolfeadam** [comment]
Without more information, it's impossible to say. If you could share the code you're calling in the `attempt` in addition to the console text (e.g., copy paste in a file and upload it here), I can try to help further.

**Comment by elmar0103** [comment]
Hi Jacob, I am currently facing the same issue. Did you already manage to solve this yourself?

**Reply by t-wolfeadam** [reply]
As I stated in my earlier comment to them, I need more information to go on: copy paste the full stacktrace and include your model if you can

**Reply by jacobtw2** [reply]
Hi. I think I understand it a bit more now. @t-wolfeadam please correct me if I am wrong

Attempt.getfeedback() basically returns whatever is being logged by the python file you are running. This is errors and warnings. 

If either of those happen then attempt.isSuccessful() returns false

If attempt.isSuccessful returns true then it essentially means nothing was returned to getFeedback() so you don't need to call it

**Reply by elmar0103** [reply]
Thank you both for your response, but I managed to get it done. What I do is send Python dataframe as string values into AnyLogic using Pypeline, and than use classes within AnyLogic to convert it into a collection. From the collection, I call the elements one by one into a source block. In this way, my production schedule I generated in Python functions as the input of my AnyLogic simulation.

**Reply by t-wolfeadam** [reply]
> Attempt.getfeedback() basically returns whatever is being logged by the python file you are running. This is errors and warnings.
> 
> If either of those happen then attempt.isSuccessful() returns false
> 
> If attempt.isSuccessful returns true then it essentially means nothing was returned to getFeedback() so you don't need to call it

@jacobtw2 Mostly correct! The last point only applies to `run` calls (for non-returning statements). The 'feedback' can hold either error messages (if any) or, for `runResults` the value that Python returns. You can see a few examples in [the class docs here](https://github.com/t-wolfeadam/AnyLogic-Pypeline/blob/master/Source/AL-Pypeline%20Library/Pypeline%20Library.alp#L4942-L4950).

That still doesn't explain how/why you got a *java* exception though, and I can't see anything in the implementation that would cause that (I'm assuming you got it sorted though)
